src/resource.ts

```diff
@@ -0,0 +1,150 @@
+import axios from "axios";
+import moment from "moment";
+import BlueButton from "./index";
+import { AuthorizationToken } from "./entities/AuthorizationToken";
+
+// initInterval in milli-seconds
+export const retrySettings = {
+  initInterval: 5000,
+  maxAttempts: 3,
+  retryableCodes: [500, 502, 503, 504],
+};
+
+// also serves as central registry for supported resource paths
+export enum FhirResourceType {
+  Patient = "fhir/Patient/",
+  Coverage = "fhir/Coverage/",
+  Profile = "connect/userinfo",
+  ExplanationOfBenefit = "fhir/ExplanationOfBenefit/",
+}
+
+function sleep(time: number) {
+  return new Promise((resolve) => {
+    setTimeout(resolve, time);
+  });
+}
+
+function isRetryable(error: any) {
+  return (
+    error.response &&
+    retrySettings.retryableCodes.includes(error.response.status)
+  );
+}
+
+async function doRetry(fhirUrl: string, config: any) {
+  let resp = null;
+
+  for (let i = 0; i < retrySettings.maxAttempts; i += 1) {
+    const waitInMilliSec = retrySettings.initInterval * 2 ** i;
+    await sleep(waitInMilliSec);
+    try {
+      resp = await axios.get(fhirUrl, config);
+      break;
+    } catch (error: any) {
+      if (error.response) {
+        resp = error.response;
+      }
+    }
+  }
+  return resp;
+}
+
+async function refreshAccessToken(
+  authToken: AuthorizationToken,
+  bb: BlueButton
+) {
+  const resp = await axios.post(
+    `${String(bb.baseUrl)}/v${bb.version}/o/token/`,
+    {},
+    {
+      auth: {
+        username: bb.clientId,
+        password: bb.clientSecret,
+      },
+      params: {
+        grant_type: "refresh_token",
+        client_id: bb.clientId,
+        refresh_token: authToken.refreshToken,
+      },
+    }
+  );
+
+  if (resp.status !== 200) {
```

Comment:
You actually don't need this. Axios will automatically throw an error for any status code outside of the 2** range.

Reply:
True.